core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpExecutionServiceDAO.scala

```diff
@@ -28,49 +29,47 @@ class HttpExecutionServiceDAO( executionServiceURL: String, submissionTimeout: F
   private implicit lazy val baseMetricBuilder =
     ExpandedMetricBuilder.expand(SubsystemMetricKey, Subsystems.Cromwell)
 
+  private def pipeline[A: FromResponseUnmarshaller](userInfo: UserInfo, retryCount: Int = 0) =
+    addAuthHeader(userInfo) ~> instrumentedGzSendReceive(retryCount) ~> unmarshal[A]
+
+  private def pipelineWithTimeout[A: FromResponseUnmarshaller](userInfo: UserInfo, timeout: Timeout, retryCount: Int = 0) =
+    addAuthHeader(userInfo) ~> instrumentedGzSendReceive(timeout, retryCount) ~> unmarshal[A]
+
   override def submitWorkflows(wdl: String, inputs: Seq[String], options: Option[String], userInfo: UserInfo): Future[Seq[Either[ExecutionServiceStatus, ExecutionServiceFailure]]] = {
     val timeout = Timeout(submissionTimeout)
     val url = executionServiceURL+"/workflows/v1/batch"
-
-    val pipeline = addAuthHeader(userInfo) ~> instrumentedGzSendReceive(timeout) ~> unmarshal[Seq[Either[ExecutionServiceStatus, ExecutionServiceFailure]]]
     val formData = Map("workflowSource" -> BodyPart(wdl), "workflowInputs" -> BodyPart(inputs.mkString("[", ",", "]"))) ++ options.map("workflowOptions" -> BodyPart(_))
-    pipeline(Post(url, MultipartFormData(formData)))
+    pipelineWithTimeout[Seq[Either[ExecutionServiceStatus, ExecutionServiceFailure]]](userInfo, timeout) apply (Post(url, MultipartFormData(formData)))
   }
 
   override def status(id: String, userInfo: UserInfo): Future[ExecutionServiceStatus] = {
     val url = executionServiceURL + s"/workflows/v1/${id}/status"
-    val pipeline = addAuthHeader(userInfo) ~> instrumentedGzSendReceive ~> unmarshal[ExecutionServiceStatus]
-    retry(when500) { () => pipeline(Get(url)) }
+    retry(when500) { count => pipeline[ExecutionServiceStatus](userInfo, count) apply Get(url) }
```

Comment:
You might want to scroll down and check out `Retry.scala`. Basically I changed it so each retry iteration passes the current retry count as an `Int`, instead of `Unit`. The retryCount is passed to `instrumentedGzSendReceive` and is instrumented with a different suffix (`.retry` instead of `.request`). This should let us see how often we're retrying requests to which services.